### PR TITLE
CATTY-526 Convert SetBrightnessBrick and ChangeBrightnessByNBrick to Swift

### PR DESCRIPTION
**Pull request by amelak9**
- [x] Include the name of the Jira ticket in the PR’s title
- [x] Verify that the Jira ticket is in the status *Ready for Development*
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s git workflow (rebase and squash your commits)
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [x] Post a message in the *#catty* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: src/Catty/DataModel/Bricks/Look/ChangeBrightnessByNBrick.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        self.changeBrightness
+    }
+
+    func setFormula(_ formula: Formula?, forLineNumber lineNumber: Int, andParameterNumber paramNumber: Int) {
+        self.changeBrightness = formula
+    }
+
+    func getFormulas() -> [Formula]! {
+        [changeBrightness!]
+    }
```
Comment (Bomml) [inline src/Catty/DataModel/Bricks/Look/ChangeBrightnessByNBrick.swift:41]: Do not force-unwrap here. guard let is a good option for example.

File: src/Catty/DataModel/Bricks/Look/SetBrightnessBrick.swift [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    func formula(forLineNumber lineNumber: Int, andParameterNumber paramNumber: Int) -> Formula? {
+        self.brightness
+    }
+
+    func setFormula(_ formula: Formula?, forLineNumber lineNumber: Int, andParameterNumber paramNumber: Int) {
+        self.brightness = formula
+    }
+
+    func getFormulas() -> [Formula]! {
+        [brightness!]
```
Comment (Bomml) [inline src/Catty/DataModel/Bricks/Look/SetBrightnessBrick.swift:40]: Do not force-unwrap (like above)

**Review by Bomml — CHANGES_REQUESTED** [review]
You deleted the CBXMLHandler files. Please convert these files also to swift. They are needed to parse the project.

Please also convert the BrickCell files.

File: src/Catty/DataModel/Bricks/Look/ChangeBrightnessByNBrick.swift [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    override required init() {
+        super.init()
+    }
+
+    func category() -> kBrickCategoryType {
+        kBrickCategoryType.lookBrick
+    }
+
+    func formula(forLineNumber lineNumber: Int, andParameterNumber paramNumber: Int) -> Formula? {
+        self.changeBrightness
```
Comment (GatCode) [inline src/Catty/DataModel/Bricks/Look/ChangeBrightnessByNBrick.swift:36]: please add the missing return

File: src/Catty/DataModel/Bricks/Look/ChangeBrightnessByNBrick.swift [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    func setFormula(_ formula: Formula?, forLineNumber lineNumber: Int, andParameterNumber paramNumber: Int) {
+        self.changeBrightness = formula
+    }
+
+    func getFormulas() -> [Formula]! {
+        [changeBrightness!]
```
Comment (GatCode) [inline src/Catty/DataModel/Bricks/Look/ChangeBrightnessByNBrick.swift:44]: return is missing and please don't use a force unwrap here - also maybe you can find a better solution for the [Formula]!

File: src/Catty/DataModel/Bricks/Look/ChangeBrightnessByNBrick.swift [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+    }
+
+    func allowsStringFormula() -> Bool {
+        false
```
Comment (GatCode) [inline src/Catty/DataModel/Bricks/Look/ChangeBrightnessByNBrick.swift:48]: return is missing

File: src/Catty/DataModel/Bricks/Look/SetBrightnessBrick.swift [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+@objcMembers class SetBrightnessBrick: Brick, BrickProtocol, BrickFormulaProtocol {
+
+    var brightness: Formula?
+
+    override required init() {
+        super.init()
+    }
+
+    func category() -> kBrickCategoryType {
+        kBrickCategoryType.lookBrick
```
Comment (GatCode) [inline src/Catty/DataModel/Bricks/Look/SetBrightnessBrick.swift:32]: returns also missing in that file

File: src/Catty/DataModel/Bricks/Look/SetBrightnessBrick.swift [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+    func formula(forLineNumber lineNumber: Int, andParameterNumber paramNumber: Int) -> Formula? {
+        self.brightness
+    }
+
+    func setFormula(_ formula: Formula?, forLineNumber lineNumber: Int, andParameterNumber paramNumber: Int) {
+        self.brightness = formula
+    }
+
+    func getFormulas() -> [Formula]! {
+        [brightness!]
```
Comment (GatCode) [inline src/Catty/DataModel/Bricks/Look/SetBrightnessBrick.swift:44]: please do not use the force unwrap here

File: src/Catty/Views/Custom/CollectionViewCells/BrickCells/Look/ChangeBrightnessByNBrickCell.swift [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+
+    static func cellHeight() -> CGFloat {
+        CGFloat(kBrickHeight2h)
```
Comment (GatCode) [inline src/Catty/Views/Custom/CollectionViewCells/BrickCells/Look/ChangeBrightnessByNBrickCell.swift:38]: returns missing

File: src/Catty/Views/Custom/CollectionViewCells/BrickCells/Look/SetBrightnessBrickCell.swift [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+    }
+
+    required init?(coder: NSCoder) {
+        super.init(coder: coder)
+    }
+
+    static func cellHeight() -> CGFloat {
+        CGFloat(kBrickHeight2h)
```
Comment (GatCode) [inline src/Catty/Views/Custom/CollectionViewCells/BrickCells/Look/SetBrightnessBrickCell.swift:38]: returns missing

File: src/Catty/DataModel/Bricks/Look/ChangeBrightnessByNBrick.swift (hunk 3 above)
Comment (GatCode) [inline src/Catty/DataModel/Bricks/Look/ChangeBrightnessByNBrick.swift:36]: return is missing a lot of files - not only that one - please check all functions again

File: src/Catty/DataModel/Bricks/Look/ChangeBrightnessByNBrick.swift (hunk 4 above)
Comment (GatCode) [inline src/Catty/DataModel/Bricks/Look/ChangeBrightnessByNBrick.swift:44]: please avoid force unwrapping here - try to find a better solution - also with the [Formula]!

File: src/Catty/DataModel/Bricks/Look/SetBrightnessBrick.swift (hunk 7 above)
Comment (GatCode) [inline src/Catty/DataModel/Bricks/Look/SetBrightnessBrick.swift:44]: please try to avoid it here too

**Review by GatCode — CHANGES_REQUESTED** [review]

**Review by GatCode — CHANGES_REQUESTED** [review]

**Review by GatCode — APPROVED** [review]

**Comment by m-herold** [comment]
Thanks!

**Pull request merged**